### ISSUE TITLE
Back to all discussions bug

### ISSUE THREAD

**Issue by zerosonesfun** [MichaelBelgium/flarum-discussion-views#17]
If you install 3.0 of this extension and install the Pages extension, go to a page and then use Flarum’s built in back button, the discussions won’t load. No errors are logged.

**Comment by MichaelBelgium** [comment]
Any javascript errors?

**Comment by zerosonesfun** [comment]
It has to do with the number abbreviation setting.

Uncaught TypeError: Cannot read property 'toString' of undefined in abbreviateNumber.js

**Comment by MichaelBelgium** [comment]
> Uncaught TypeError: Cannot read property 'toString' of undefined in abbreviateNumber.js

Does any of the discussions have 0 views? Or does it happen either always? This is the least expected issue by far..

**Comment by zerosonesfun** [comment]
> > Uncaught TypeError: Cannot read property 'toString' of undefined in abbreviateNumber.js
> 
> Does any of the discussions have 0 views? Or does it happen either always? This is the least expected issue by far..

Happens either way. I still think there’s something about the way the discussion list loads after going to a page which is a little different from the way the discussion list loads other times. 

If I go to a discussion and the use the built in back button, everything is fine and I also noticed that the discussion list loads instantly, no loading spinner at all. However, if you go back from a page, even without this extension installed, there’s a loading spinner at least for a second, and then the discussions load.

In other words, pages and possibly certain other places in Flarum, must use slightly different code to go back to the all discussions list compared to how it is done if you are viewing a discussion. 

This slight difference in the way the all discussions list is sometimes loaded may be causing something such as, the JavaScript to not load at the right moment or something like that. That’s the only theory I have. 

A quick Google of this error gives a possible solution having to do with adding code to ignore null or undefined. But, is the abbreviatedNumber JavaScript part of Flarum? I don’t see it in this repository. If it’s already a part of Flarum then this is even more curious because the error only happens with this extension.

**Comment by MichaelBelgium** [comment]
> However, if you go back from a page, even without this extension installed, there’s a loading spinner at least for a second, and then the discussions load.

Like what page?

> But, is the abbreviatedNumber JavaScript part of Flarum? I don’t see it in this repository. If it’s already a part of Flarum then this is even more curious because the error only happens with this extension.

Yeah the `abbreviatedNumber.js` is from flarum itself, it accepts a number which then transforms it. In your use case it passes an undefined variable through, then it says

> Uncaught TypeError: Cannot read property 'toString' of undefined in abbreviateNumber.js

The thing is, it's [always a number](https://github.com/MichaelBelgium/flarum-discussion-views/blob/master/js/src/forum/components/AddViewsToModelAndDisplay.js#L17) that get passed through. `.length` can't return an undefined number, its either 0 or more (counting the views of a discussion in the table `discussion_views`)

If only i can see it too :/
